master/internal/experiment/bulk_action.go

```diff
@@ -817,6 +817,58 @@ func MoveExperiments(
 			}
 		}
 
+		if _, err = tx.NewRaw(`
+		UPDATE runs SET local_id=s.local_id
+		FROM
+			(
+				SELECT
+					r.id as id,
+					(p.max_local_id + ROW_NUMBER() OVER(PARTITION BY p.id ORDER BY r.id)) as local_id
+				FROM
+					projects p
+					JOIN runs r ON r.project_id=p.id
+				WHERE r.experiment_id IN (?)
+			) as s 
+		WHERE s.id=runs.id
+		`,
+			bun.In(acceptedIDs)).Exec(ctx); err != nil {
+			return nil, fmt.Errorf("updating run's local IDs: %w", err)
+		}
+
+		if _, err = tx.NewRaw(`
+		UPDATE projects SET max_local_id=s.max_local_id
```

Comment:
`max_local_id` represents the highest id value that has ever been used rather than the highest value currently being used. For example if `max_local_id` is 5 and we delete the run with the local id of 5, we do not lower `max_local_id`. It is a strictly increasing value. We never want to re-use a local id. It also means that we cannot compute it by just checking `MAX(local_id)` in the run table.

Reply:
Ah, yes, thank you! That's a good reason I missed.